File: domain/src/main/java/com/beat/domain/schedule/repository/ScheduleRepository.java
package com.beat.domain.schedule.repository;

import com.beat.domain.schedule.domain.Schedule;
import com.beat.domain.schedule.repository.dto.MinPerformanceDateDto;
import java.util.List;
import java.util.Optional;

public interface ScheduleRepository {
    Optional<Schedule> findById(Long id);

    Optional<Schedule> lockById(Long id);

    List<Schedule> findAllByPerformanceId(Long performanceId);

    List<Long> findIdsByPerformanceId(Long performanceId);

    int countByPerformanceId(Long performanceId);

    Schedule save(Schedule schedule);

    List<Schedule> saveAll(List<Schedule> schedules);

    void delete(Schedule schedule);

    void deleteByPerformanceId(Long performanceId);

    List<Schedule> findPendingSchedules();

    List<MinPerformanceDateDto> findMinPerformanceDateByPerformanceIds(List<Long> performanceIds);
}

File: domain/src/main/kotlin/com/beat/domain/schedule/repository/ScheduleRepository.kt
package com.beat.domain.schedule.repository

import com.beat.domain.schedule.domain.Schedule
import java.util.*

@JvmSuppressWildcards
interface ScheduleRepository {
    fun findById(id: Long?): Optional<Schedule>

    fun lockById(id: Long?): Optional<Schedule>

    fun findAllByPerformanceId(performanceId: Long?): List<Schedule>

    fun findAllById(ids: Collection<Long>): List<Schedule>

    fun findIdsByPerformanceId(performanceId: Long?): List<Long>

    fun countByPerformanceId(performanceId: Long?): Int

    fun save(schedule: Schedule): Schedule

    fun saveAll(schedules: List<Schedule>): List<Schedule>

    fun delete(schedule: Schedule)

    fun deleteByPerformanceId(performanceId: Long?)

    fun findPendingSchedules(): List<Schedule>
}
